BooruSharp.Others/Pixiv.cs

```diff
@@ -128,108 +92,86 @@ public async Task<byte[]> PreviewToByteArrayAsync(SearchResult result)
             var request = new HttpRequestMessage(HttpMethod.Get, result.PreviewUrl);
             request.Headers.Add("Referer", result.PostUrl.AbsoluteUri);
 
-            var response = await HttpClient.SendAsync(request);
+            var response = await GetResponseAsync(request);
 
             response.EnsureSuccessStatusCode();
 
             return await response.Content.ReadAsByteArrayAsync();
         }
 
         /// <summary>
-        /// Checks if the <see cref="AccessToken"/> needs to be updated,
+        /// Checks if the authentication token needs to be updated,
         /// and updates it if needed.
         /// </summary>
+        /// <remarks>
+        /// You must login using <see cref="Auth"/> property before calling this method.
+        /// </remarks>
         /// <returns>The task object representing the asynchronous operation.</returns>
         /// <exception cref="AuthentificationInvalid"/>
+        /// <exception cref="AuthentificationRequired"/>
         /// <exception cref="HttpRequestException"/>
+        // TODO: does this have to be public?
         public async Task CheckUpdateTokenAsync()
```

Comment:
Same here. I don't think this has to be explicitly called by the user.

Reply:
Iirc this was used when you provide your own token

Reply:
But since all the other methods call this method to validate the auth info as well as update the tokens, does it make sense to make `CheckUpdateTokenAsync` `private`?

Reply:
Speaking of that, looks like the favorite functions don't do that, that may be a mistake from me